facebook-commerce-events-tracker.php

```diff
@@ -178,84 +182,116 @@ function( $item ) {
+				add_action( 'woocommerce_before_shop_loop', [ $this, 'actually_inject_search_event' ] );
 			}
 		}
 
+
 		/**
-		 * Triggers Search for result pages
+		 * Sends a server-side Search event.
+		 *
+		 * @internal
+		 *
+		 * @since 2.0.0-dev.3
 		 */
-		public function actually_inject_search_event() {
+		public function send_search_event() {
+
+			$this->send_api_event( $this->get_search_event() );
+		}
+
+
+		/**
+		 * Creates an Event instance to track a search request.
+		 *
+		 * The event instance is stored in memory to return a single instance per request.
+		 *
+		 * @since 2.0.0-dev.3
+		 *
+		 * @return Event
+		 */
+		private function get_search_event() {
 			global $wp_query;
 
-			if ( ! self::$isEnabled || empty( $_GET['post_type'] ) || 'product' !== $_GET['post_type'] ) {
-				return;
-			}
+			if ( null === $this->search_event ) {
 
-			// if any product is a variant, fire the pixel with
-			// content_type: product_group
-			$content_type = 'product';
-			$product_ids  = [];
-			$contents     = [];
-			$total_value  = 0.00;
+				// if any product is a variant, fire the pixel with content_type: product_group
+				$content_type = 'product';
+				$product_ids  = [];
+				$contents     = [];
+				$total_value  = 0.00;
 
-			foreach ( $wp_query->posts as $post ) {
+				foreach ( $wp_query->posts as $post ) {
 
-				$product = wc_get_product( $post );
+					$product = wc_get_product( $post );
 
-				if ( ! $product instanceof \WC_Product ) {
-					continue;
-				}
+					if ( ! $product instanceof \WC_Product ) {
+						continue;
+					}
 
-				$product_ids = array_merge( $product_ids, WC_Facebookcommerce_Utils::get_fb_content_ids( $product ) );
+					$product_ids = array_merge( $product_ids, WC_Facebookcommerce_Utils::get_fb_content_ids( $product ) );
 
-				$contents[] = [
-					'id'       => \WC_Facebookcommerce_Utils::get_fb_retailer_id( $product ),
-					'quantity' => 1, // consider the search results a quantity of 1
-				];
+					$contents[] = [
+						'id'       => \WC_Facebookcommerce_Utils::get_fb_retailer_id( $product ),
+						'quantity' => 1, // consider the search results a quantity of 1
+					];
 
-				$total_value += (float) $product->get_price();
+					$total_value += (float) $product->get_price();
 
-				if ( WC_Facebookcommerce_Utils::is_variable_type( $product->get_type() ) ) {
-					$content_type = 'product_group';
+					if ( WC_Facebookcommerce_Utils::is_variable_type( $product->get_type() ) ) {
+						$content_type = 'product_group';
+					}
 				}
+
+				$event_data = [
+					'event_name'  => 'Search',
+					'custom_data' => [
+						'content_type'  => $content_type,
+						'content_ids'   => json_encode( array_slice( $product_ids, 0, 10 ) ),
+						'contents'      => $contents,
+						'search_string' => get_search_query(),
+						'value'         => \SkyVerge\WooCommerce\PluginFramework\v5_5_4\SV_WC_Helper::number_format( $total_value ),
```

Comment:
can we add a `use` at the beginning of the file like it was done for `use SkyVerge\WooCommerce\Facebook\Events\Event;` so were the helper method can be written as `Framework\SV_WC_Helper::...`

Reply:
Yes, fixed in 9db9c5b.